tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py

```diff
@@ -35,46 +47,50 @@ def test_verify_parsing_of_twilio_message(event):
     assert response, "The dictionary should not be empty"
     assert 'AddOns' not in response
 
-
 @pytest.mark.parametrize('event', [(albInvokedWithoutAddOn), (albInvokeWithAddOn), (sqsInvokedWithAddOn)])
 def test_request_makes_vetext_call(mocker, http_success_response, event):
-    vetext_mock = mocker.patch(f'{lambda_module}.make_vetext_request', return_value=http_success_response)
+    sqs_mock = mocker.patch(f'{lambda_module}.push_to_sqs')
+    mocker.patch(f'{lambda_module}.read_from_ssm', return_value="ssm")
+    mocker.patch.dict(os.environ, {"vetext_api_endpoint_domain": "some.domain", "vetext_api_endpoint_path": "/some/path"})
+    mocker.patch("http.client.HTTPSConnection", return_value=MockConnection(http_success_response))    
```

Comment:
Your tests are passing, so this seems to work.  Rather than mocking the connection as a class, I would just mock the request function to do nothing and the response function to return whatever value you want.  This is what you did with your sqs_mock.

Ditto for subsequent uses.

Reply:
I'm not sure how I would mock the request/response without separating out the request/response handling into other functions.  Mocking the connection and specifying the response allows me to mock any request/response without changing anything.  

For now I will keep things as is, but I'd like to learn more about your suggested approach.